### DISCUSSION TITLE
property with default value causes entity to change the state and perform an update on next flush()

### DISCUSSION THREAD

**Discussion by dginzbourg** [mikro-orm/mikro-orm#4284] [Q&A]
**Describe the bug**
I've an entity like:
```
@Entity()
export class AEntity {
  @PrimaryKey()
  id!: string;

  @Property({ default: 'value1', defaultRaw: 'value1' })
  created_at?: string;
}
```
And the code snippet shows the sequence when flush() will be called again (e.g. different entity was created, etc):
```
    const expDto = { id: '1' };
    const entity = await orm.em.create(AEntity, expDto);
    await orm.em.persistAndFlush(entity);
    // await orm.em.clear();
    await orm.em.flush();
```
The generated sqls are:
```
[query] commit
[query] insert into `aentity` (`id`) values ('1') returning `id`, `created_at` [took 0 ms]
[query] update `aentity` set `created_at` = 'value1' where `id` = '1' [took 0 ms]
[query] commit
```

**Stack trace**
```
...
```

**To Reproduce**
1. ...
2. ...
3. ...

**Expected behavior**
I'd assume that the "returning" properties won't cause unit-of-work change. Is it intentional? Maybe I'm missing something in the property definition that would tweak the logic? 
We have BaseEntity that has created_at, updated_at properties that are updated by DB. And we found out that there're extra updates generated on subsequent flush() that logically we didn't expect (using the same logic as for PrimaryKey - generated ID doesn't leave entity as a "changed" after flush())
I can provide full working example, just before that wanted to understand if it's expected behavior.

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 18.3.0 |
| typescript | 4.8 |
| mikro-orm | 5.7.2|
| your-driver | sqlite (for testing) postgresql (my app)|

**Comment by B4nan** [comment]
>  `@Property({ default: 'value1', defaultRaw: 'value1' })`

This seems to be wrong, `defaultRaw` needs to be quoted if it is supposed to be a string value. Providing the same value for both will work only with a number or boolean as those are not quoted. There is no need to use them both, `defaultRaw` is meant for stuff like SQL functions, e.g. `now()`, you don't need that for scalars. Also note that both `em.create` and `em.clear` are synchronous methods, no need to await them (although that itself won't alter any behavior, you are free to await anything in JS).

But even when I fix that bit, I am still able to reproduce this, I can see the entity state is not updated correctly after flushing. Let me have a closer look, this is indeed wrong and needs to be fixed ASAP.

**Reply by dginzbourg** [reply]
Yes, I'm using `defaultRaw: 'now()'` in postgres. Tried to make something reproducible with sqlite. Thank you for promptly response.

**Reply by B4nan** [reply]
This should do it: https://github.com/mikro-orm/mikro-orm/commit/e1230764633ecdc906f3b1a4073f1dc44100e5a1

New dev version will be out in ~15 minutes, will try to ship stable later this evening.
